test/metabase/query_processor/card_test.clj

```diff
@@ -165,3 +169,40 @@
                                                                             :some_other_key  [:ref [:field Integer/MAX_VALUE {:base-type :type/DateTime, :temporal-unit :month}]]}}}}]
       (is (= [[100]]
              (mt/rows (run-query-for-card card-id)))))))
+
+(deftest nested-query-permissions-test
```

Comment:
This test is mostly the same as the other one but uses `qp.card/process-query-for-card` instead of hitting the REST API endpoint, I was on the fence about which one was better so I just ended up putting them both in here. The REST API test is good so we can be sure that it really does work end-to-end and doesn't fail some other perms check outside of QP land, while the QP-specific test is good so we can be confident the QP is working in isolation from the REST API (e.g. if we only run QP tests with the `:test/qp` alias)